src/wire_msg.rs

```diff
@@ -0,0 +1,144 @@
+// Copyright 2021MaidSafe.net limited.
+//
+// This SAFE Network Software is licensed to you under the MIT license <LICENSE-MIT
+// https://opensource.org/licenses/MIT> or the Modified BSD license <LICENSE-BSD
+// https://opensource.org/licenses/BSD-3-Clause>, at your option. This file may not be copied,
+// modified, or distributed except according to those terms. Please review the Licences for the
+// specific language governing permissions and limitations relating to use of the SAFE Network
+// Software.
+
+use super::{Error, MsgEnvelope, Result};
+use cookie_factory::{bytes::be_u16, combinator::slice, gen};
+use std::fmt::Debug;
+
+/// Current version of the messaging protocol
+const MESSAGING_PROTO_VERSION: u16 = 1u16;
```

Comment:
thought we are supposed to use u32 ?

or you mean use 16 for version, and 16 for PayloadSerType  ?

Reply:
correct, 2 bytes for each field, i suspect is enough?